src/agents/interactive_parameters.jl

```diff
@@ -109,36 +108,45 @@ function abm_data_exploration!(fig, abmstepper, model, agent_step!, model_step!,
             end
         end
     end
-    
-    # Clicking the run button
+
+    # Clicking the "run" button
     isrunning = Observable(false)
-    on(run) do clicks; isrunning[] = !isrunning[]; end
+    on(run) do clicks
+        isrunning[] = !isrunning[]
+    end
     on(run) do clicks
         @async while isrunning[]
-            n = speed[]
-            model = modelobs[] # this is useful only for the reset button
-            Agents.step!(abmstepper, model, agent_step!, model_step!, n)
-            if L > 0
-                s += n
-                if L > 0 && Agents.should_we_collect(s, model, when) # update collected data
-                    push!(N.val, s)
-                    update_abm_data_plots!(data, axs, model, adf, mdf, adata, mdata, N)
-                end
-            end
+            step[] = step[] + 1
             slep[] == 0 ? yield() : sleep(slep[])
             isopen(fig.scene) || break # crucial, ensures computations stop if closed window.
         end
     end
 
-    # Clicking the reset button
+    # Clicking the "reset" button
     on(reset) do clicks
         modelobs[] = deepcopy(model0)
+        model = modelobs[]
         Agents.step!(abmstepper, model, agent_step!, model_step!, 0)
         L > 0 && add_reset_line!(axs, s)
-        update[] = update[] + 1 # also trigger parameter updates
+        update[] = update[] + 1 # trigger update button
     end
-    
-    # Clicking the update button
+
+    # Clicking the "clear all" button
+    on(clear) do clicks
+        reset[] = reset[] + 1 # trigger reset button
+        # empty!([adf, mdf])
+        N[] = [0] # set new starting point for plots
+        s = 0 # reset current step
+        if L > 0
+            for ax in axs
+                delete!(ax)
+            end
+            data, axs = init_abm_data_plots!(fig, datalayout, model, adf, mdf,
+                adata, mdata, N, alabels, mlabels, colorscheme)
+        end
+    end
```

Comment:
This is the core functionality of the newly added button. I've tested it, of course, but not extensively.

Reply:
I'd like to get some feedback from you regarding line 137. My initial idea was that we could also delete all collected data so far when the "clear all" button is clicked. But I realised that it should really only reset the view and not the underlying dataframes of previous runs made in the same window. Do you have an opinion on this?